Assets/MixedRealityToolkit/_Core/Definitions/InputSystem/InteractionDefinition.cs

```diff
@@ -0,0 +1,195 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Internal.Definitions.InputSystem
+{
+    /// <summary>
+    /// Maps the capabilities of controllers, one definition should exist for each interaction profile.<para/>
+    /// <remarks>Interactions can be any input the controller supports such as buttons, triggers, joysticks, dpads, and more.</remarks>
+    /// </summary>
+    public struct InteractionDefinition
+    {
+        public InteractionDefinition(uint id, AxisType axisType, InputType inputType) : this()
+        {
+            Id = id;
+            AxisType = axisType;
+            InputType = inputType;
+        }
+
+        #region Interaction Properties
+
+        /// <summary>
+        /// The Id assigned to the Interaction.
+        /// </summary>
+        public uint Id { get; }
+
+        /// <summary>
+        /// The axis type of the button, e.g. Analogue, Digital, etc.
+        /// </summary>
+        public AxisType AxisType { get; }
+
+        /// <summary>
+        /// The primary action of the input as defined by the controller SDK.
+        /// </summary>
+        public InputType InputType { get; }
+
+        /// <summary>
+        /// Action to be raised to the Input Source when the input data has changed.
+        /// </summary>
+        public InputAction InputAction { get; set; }
+
+        /// <summary>
+        /// Has the value changed since the last reading.
+        /// </summary>
+        public bool Changed { get; private set; }
+
+        #endregion Interaction Properties
+
+        #region Definition Data items
+
+        /// <summary>
+        /// The data storage for a Raw / None Axis type.
+        /// </summary>
+        private object rawData;
+
+        /// <summary>
+        /// The data storage for a Digital Axis type.
+        /// </summary>
+        private bool boolData;
+
+        /// <summary>
+        /// The data storage for a Single Axis type.
+        /// </summary>
+        private float floatData;
+
+        /// <summary>
+        /// The data storage for a Dual Axis type.
+        /// </summary>
+        private Vector2 vector2Data;
+
+        /// <summary>
+        /// The position data storage for a 3DoF type.
+        /// </summary>
+        private Vector3 positionData;
+
+        /// <summary>
+        /// The rotation data storage for a 3DoF type.
+        /// </summary>
+        private Quaternion rotationData;
+
+        private Tuple<Vector3, Quaternion> transformData;
+
+        #endregion Definition Data items
```

Comment:
you were supposed to leave in the single Get, but I'll re-add this after merge

Reply:
Ah, my bad.